core/src/main/java/org/apache/calcite/avatica/MetaImpl.java

```diff
@@ -388,6 +388,32 @@ public MetaColumn(
     public String getName() {
       return columnName;
     }
+    public static String[] getColumnNames() {
```

Comment:
Why `String[]` instead of `List<String>`? Why not return a `Collections.unmodifiableList` stored in a static readonly field? This news up a list and three arrays (including one for the varargs) and does two copies every time it's called, absent JVM optimizations. Not the end of the world, but it's easy to reduce the overhead if you can use Lists instead of arrays, as Effective Java and almost everyone else advises.

Reply:
The reason why I kept that was because of how `createResultSet` is set up to take in the varags.
I could restructure `createResultSet` to work with `List<String>`

What would be the best way to instantiate this when considering the number of copies?

How is: 
```
public static final List<String> COLUMN_NAMES = Collections.unmodifiableList(
      Arrays.asList("TABLE_CAT",
          "TABLE_SCHEM",
          "TABLE_NAME",
          "TABLE_TYPE",
          "REMARKS",
          "TYPE_CAT",
          "TYPE_SCHEM",
          "TYPE_NAME",
          "SELF_REFERENCING_COL_NAME",
          "REF_GENERATION"));

Reply:
So if it's most ergonomic as a string array, then just new up a string array directly rather than going through asList -> toArray. As in `return new String[] { "TABLE_CAT", "TABLE_SCHEM", ... };`